frontend/packages/ceph-storage-plugin/src/components/independent-mode/install.tsx

```diff
@@ -0,0 +1,266 @@
+import * as _ from 'lodash';
+import { match } from 'react-router';
+import {
+  ButtonBar,
+  NsDropdown,
+  withHandlePromise,
+  HandlePromiseProps,
+} from '@console/internal/components/utils';
+import { k8sGet, k8sCreate, referenceForModel } from '@console/internal/module/k8s';
+import { ClusterServiceVersionModel } from '@console/operator-lifecycle-manager';
+import { Title, FormGroup, Form, ActionGroup, Button, TextInput } from '@patternfly/react-core';
+import { history } from '@console/internal/components/utils/router';
+import { SecretModel, ConfigMapModel } from '@console/internal/models';
+import { getName } from '@console/shared';
+import { OCSServiceModel } from '../../models';
+import FileUpload from './fileUpload';
+import { DataState, ErrorType, Field } from './types';
+import { getValidJSON, checkError } from './utils';
+import './install.scss';
+
+const ERROR: DataState = {
+  ns: '',
+  fsid: '',
+  admin: '',
+  mondata: '',
+};
+
+type ErrorTextProps = {
+  text: string;
+};
+
+const ErrorText: React.FC<ErrorTextProps> = ({ text }) => (
+  <span className="im-install-page--error">{text}</span>
+);
+
+const getErrorText = (text: string) => <ErrorText text={text} />;
+
+const InstallExternalCluster = withHandlePromise((props: InstallExternalClusterProps) => {
+  const {
+    inProgress,
+    errorMessage,
+    handlePromise,
+    match: {
+      params: { ns, appName },
+    },
+  } = props;
+  const [clusterServiceVersion, setClusterServiceVersion] = React.useState(null);
+  const [fileData, setFileData] = React.useState('');
+  const [fileName, setFileName] = React.useState('');
+  const [namespace, setNamespace] = React.useState(ns);
+  const [externalFSID, setExternalFSID] = React.useState(null);
+  const [externalAdminSecret, setExternalAdminSecret] = React.useState(null);
+  const [externalMonData, setExternalMonData] = React.useState(null);
+  const [dataError, setDataError] = React.useState(ERROR);
+  const [fileError, setFileError] = React.useState('');
+  const [, updateState] = React.useState();
+
+  // Todo(bipuladh): React does shallow comparison dataError and fileError need deep comparison.
+  const forceUpdate = React.useCallback(() => updateState({}), []);
+
+  const setErrors = React.useCallback((errors: ErrorType[]): void => {
+    for (const err of errors) {
+      setDataError(Object.assign(ERROR, { [err.field]: err.message }));
+    }
+  }, []);
+
+  const getState = React.useCallback(
+    () => ({
+      [Field.NAMESPACE]: namespace,
+      [Field.FSID]: externalFSID,
+      [Field.ADMIN]: externalAdminSecret,
+      [Field.MONDATA]: externalMonData,
+    }),
+    [externalAdminSecret, externalFSID, externalMonData, namespace],
+  );
+
+  React.useEffect(() => {
+    setErrors(checkError(getState()));
+  }, [getState, setErrors]);
+
+  // File Upload handler
+  const onUpload = (event) => {
+    event.preventDefault();
+    const reader = new FileReader();
+    const file = event.target.files[0];
```

Comment:
```suggestion
    const file = event.target?.files[0];
```

Reply:
not required

Reply:
Declare the type for the files and reader.

Reply:
type inference is possible so the inferred type for file is string | arraybuffer